### PR TITLE
Added testcases for the resource detection processor

### PR DESCRIPTION
**Pull request by jefchien**
Bumped up the Otel SDK versions for the spark sample app.

Templates based on attributes listed in [documentation](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor) for the resource detection processor. Currently validating based on existence of attribute in trace.

Will need to add the new testcases to the CI [config](https://github.com/aws-observability/aws-otel-collector/blob/main/e2etest/testcases.json).

Tested by running terraform/imagebuild and running the terraform/ec2, terraform/ecs, and terraform/eks tests locally.

### CODE REVIEW

File: sample-apps/spark/build.gradle [hunk 1]
```diff
@@ -38,9 +38,9 @@ dependencies {
     implementation "io.grpc:grpc-api:1.34.1"
     implementation "io.grpc:grpc-netty-shaded:1.34.1"
 
-    implementation platform("io.opentelemetry:opentelemetry-bom:1.0.1")
+    implementation platform("io.opentelemetry:opentelemetry-bom:1.4.1")
     implementation "io.opentelemetry:opentelemetry-api"
-    implementation "io.opentelemetry:opentelemetry-sdk-metrics:1.0.0-alpha"
+    implementation "io.opentelemetry:opentelemetry-sdk-metrics:1.4.1-alpha"
```
Comment (mxiamxia) [inline sample-apps/spark/build.gradle]: I know there are some metric related breaking changes after OTel 1.3.x SDK. Could you help to run `otlp-metric ` test case before merging the change?

Comment (mxiamxia) [reply sample-apps/spark/build.gradle]: Or can we split this PR. let's put the SDK upgrade into a separate PR?

**Comment by mxiamxia** [comment]
Could you please have a to be done list in this PR description to let people know what are things left for finishing resource detection test cases? Eg, validator updates. push sample app image. etc.

File: sample-apps/spark/build.gradle (hunk 1 above)
Comment (jefchien) [reply sample-apps/spark/build.gradle]: Sounds good. I'll separate and test it.

**Review by mxiamxia — APPROVED** [review]
LGTM! Thanks

**Pull request merged**